_src/pattern-library/sass/components/_grid.scss

```diff
@@ -0,0 +1,357 @@
+// ------------------------------
+// edX Pattern Library: Components - Grid
+
+// About: Contains base settings and options for the default Pattern Library grid system.
+// ----------------------------
+
+// #CONFIG
+// #GLOBALS
+// #UTILITY
+// #CANNED
+// #CANNED - RESPONSIVE
+// #GALLERY
+// #GALLERY - CANNED
+// #GALLERY - CANNED + RESPONSIVE
+// #DEBUG
+
+// ----------------------------
+
+// ----------------------------
+// #CONFIG
+// ----------------------------
+// susy-based configuration (see http://susydocs.oddbird.net/en/latest/settings/)
+$susy: (
+    use-custom: (
+        background-options: false,
+        box-sizing: false,
+        breakpoint: true,
+        clearfix: true,
+        rem: true
+    ),
+    math: fluid,
+    output: float
+);
+
+// LAYOUT: set default susy settings
+// (gutters are handled outside of elements - by default)
+$susy-default: (
+    flow: $grid-direction-default,
+    column-width: $grid-column-width,
+    columns: $grid-column-count,
+    container-position: $grid-container-position,
+    container: $grid-max-width,
+    global-box-sizing: border-box,
+    gutters: $grid-gutter-width,
+    gutter-position: $grid-gutter-position
+);
+
+// LAYOUT: set susy settings especially for manually set grids in the DOM
+// (gutters are handled inside of elements - by default)
+$susy-manual: layout($susy-default $grid-manual-gutter-position);
+
+// LAYOUT: set debug settings
+$susy-debug: (
+    debug: (
+        color: $grid-debug-color,
+        image: show
+    )
+);
+
+
+// ----------------------------
+// #GLOBALS
+// ----------------------------
+// set basic grid settings for app
+@include layout($susy-default);
+
+
+// ----------------------------
+// #UTILITY
+// ----------------------------
+%grid-container {
+    @include container();
+}
+
+// grid rows - set visual break and document flow clear (clearfix)
+%grid-row {
+    @include break();
+}
+
+%grid-gallery {
+    @include break();
+}
+
+.grid-container {
+    @extend %grid-container;
+
+    .col.first {
+        @include first();
+    }
+
+    .col.last {
+        @include last();
+    }
+
+    .col.break {
+        @include break();
+    }
+
+    .col.omega {
+        @include omega();
+    }
+    .col.full {
+        @include full();
+    }
+}
+
+.row {
+    @extend %grid-row;
+}
+
+// break grid's columns to a new line
+.grid-break {
+    @include break();
+}
+
+// don't break the columns to a new line
+.grid-unbreak {
+    @include nobreak();
+}
+
+
+// ----------------------------
+// #CANNED
+// ----------------------------
+.grid-manual {
+    @include layout($susy-manual, clean);
+
+    // column widths
+    @for $i from 1 through $grid-column-count {
```

Comment:
![giphy](https://cloud.githubusercontent.com/assets/163763/8678226/6bddd540-2a21-11e5-8716-1430dbb3d6d9.gif)


Just fighting off redundancy (and sharks) in our codebase.